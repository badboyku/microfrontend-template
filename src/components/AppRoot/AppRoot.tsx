import { Outlet, useLoaderData } from 'react-router-dom';
import AppSettingsEditor from 'components/AppSettingsEditor';
import logger from 'utils/logger';
import settings from 'utils/settings';

const AppRoot = () => {
  const { isAuthorized } = useLoaderData() as { isAuthorized: boolean };
  logger.debug('microfrontend-template: components/AppRoot called', { isAuthorized });

  return (
    <>
      {!settings.isProd && <AppSettingsEditor />}
      <Outlet />
    </>
  );
};

export default AppRoot;
