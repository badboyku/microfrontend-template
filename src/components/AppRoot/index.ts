/* istanbul ignore file */
import AppRoot from './AppRoot';

export default AppRoot;
