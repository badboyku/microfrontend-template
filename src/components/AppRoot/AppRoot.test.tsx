import { render } from '@testing-library/react';
import AppRoot from './AppRoot';

const renderAppRoot = (props = {}) => {
  return render(<AppRoot {...props} />);
};

describe('Component AppRoot', () => {
  it('renders without crashing', () => {
    const { asFragment } = renderAppRoot();

    expect(asFragment()).toMatchSnapshot();
  });
});
