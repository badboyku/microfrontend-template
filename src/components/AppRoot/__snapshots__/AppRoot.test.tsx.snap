// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component AppRoot renders without crashing 1`] = `<DocumentFragment />`;
